platform/darwin/src/MGLMapSnapshotter.mm

```diff
@@ -146,6 +149,35 @@ - (void)startWithQueue:(dispatch_queue_t)queue completionHandler:(MGLMapSnapshot
     dispatch_async(queue, ^{
         _snapshotCallback = std::make_unique<mbgl::Actor<mbgl::MapSnapshotter::Callback>>(*mbgl::Scheduler::GetCurrent(), [=](std::exception_ptr mbglError, mbgl::PremultipliedImage image, mbgl::MapSnapshotter::Attributions attributions, mbgl::MapSnapshotter::PointForFn pointForFn) {
             _loading = false;
+            
+            NSMutableArray *infos = [NSMutableArray array];
+            
+            
+#if TARGET_OS_IPHONE
+            CGFloat fontSize = [UIFont smallSystemFontSize];
+            UIColor *attributeFontColor = [UIColor blackColor];
+#else
+            CGFloat fontSize = [NSFont systemFontSizeForControlSize:NSMiniControlSize];
+            NSColor *attributeFontColor = [NSColor blackColor];
+#endif
+            for (auto attribution = attributions.begin(); attribution != attributions.end(); ++attribution) {
+                NSString *attributionHTMLString = @(attribution->c_str());
+                NSArray *tileSetInfos = [MGLAttributionInfo attributionInfosFromHTMLString:attributionHTMLString
+                                                                                  fontSize:fontSize
+                                                                                 linkColor:attributeFontColor];
+                [infos growArrayByAddingAttributionInfosFromArray:tileSetInfos];
+            }
+            
+            CGFloat attributionBackgroungWidth = 0.0;
+            CGFloat attributionBackgroungHeight = 0.0;
+            for (MGLAttributionInfo *info in infos) {
+                if (info.isFeedbackLink) {
+                    continue;
+                }
+                attributionBackgroungWidth += [info.title size].width + 10;
+                attributionBackgroungHeight = [info.title size].height;
```

Comment:
If a shorter info follows a taller info, `attributionBackgroundHeight` will wind up with the shorter info’s height. `MAX()` would address that scenario.